cypress/e2e/signUp.cy.js

```diff
@@ -1,12 +1,58 @@
 /// <reference types='cypress' />
 /// <reference types='../support' />
 
+import SignUpPageObject from '../support/pages/signUp.pageObject';
+import HomePageObject from '../support/pages/home.pageObject';
+
+const signUpPage = new SignUpPageObject();
+const homePage = new HomePageObject();
+
 describe('Sign Up page', () => {
-  before(() => {
+  beforeEach(() => {
+    cy.clearCookies();
+    cy.clearLocalStorage();
+  });
+  let user;
 
+  before(() => {
+    cy.task('db:clear');
+    cy.task('generateUser').then((generateUser) => {
+      user = generateUser;
+    });
   });
 
-  it('should ...', () => {
+  it('should register new user with all filled fields', () => {
+    signUpPage.visit();
+    signUpPage.typeUsername(user.username);
+    signUpPage.typeEmail(user.email);
+    signUpPage.typePassword(user.password);
+    signUpPage.clickSignUpBtn();
+    signUpPage.clickOkBtn();
+    homePage.assertHeaderContainUsername(user.username);
+  });
 
+  it('shouldn\'t register new user with empty "username" field', () => {
+    signUpPage.visit();
+    signUpPage.typeEmail(user.email);
+    signUpPage.typePassword(user.password);
+    signUpPage.clickSignUpBtn();
+    signUpPage.clickOkBtn();
+    homePage.assertHeaderDoesntContainUsername(user.username);
+  });
+  it('shouldn\'t register new user with empty "email" field', () => {
+    signUpPage.visit();
+    signUpPage.typeUsername(user.username);
+    signUpPage.typePassword(user.password);
+    signUpPage.clickSignUpBtn();
+    signUpPage.clickOkBtn();
```

Comment:
The `clickOkBtn` method might not be necessary for this negative test case since the registration should fail and the confirmation dialog might not appear. Consider verifying if this step is needed.